src/material-moment-adapter/adapter/index.ts

```diff
@@ -35,11 +35,37 @@ export * from './moment-date-formats';
 export class MomentDateModule {}
 
 @NgModule({
-  imports: [MomentDateModule],
-  providers: [{provide: MAT_DATE_FORMATS, useValue: MAT_MOMENT_DATE_FORMATS}],
+  providers: [provideMomentDateAdapter()],
 })
 export class MatMomentDateModule {}
 
+/**
```

Comment:
We generally don't put longer docs like this in the doc string, because the API docs are already pretty busy on Material. This is already documented in the datepicker.md.

Reply:
Understandable. 

Should I update this PR to only have the NgModule providers refactor or just close it?

Reply:
The fix for the providers is okay, but I think that we can do the same for the Luxon and date-fns adapters.